### PR TITLE
Add Awaiting author contribution

### PR DESCRIPTION
**Pull request by BastianZim**
I don't see a reason why we shouldn't test for it, do you know one?

Edit: it could be that there author responds and immediately pings the team again but if this is not added the action is triggered quite often. To safe some concurrency, I would rather have that the author needs to comment twice.

### CODE REVIEW

**Review by carterbox — COMMENTED** [review]
It makes sense to not add a team label when "review-requested" or "Awaiting-author" labels are already present because their presence means the team label already exists. The state diagram for the current flow would be something like this:

```mermaid
stateDiagram-v2
    [*] --> reviewrequested,team: User pings team
    team --> reviewrequested,team: User pings team
    reviewrequested,team --> awaitingauthor,team: staged-recipes reviews
    awaitingauthor,team --> team: Any author action
```
Note that each transition between states implies a GitHub action job. 

If we want to require the author to always actively ask for a review. Is it more efficient to have a flow like below? The number of states is less and I think the number of unique jobs becomes 2.

```mermaid
stateDiagram-v2
    [*] --> reviewrequested,team: User pings team
    reviewrequested,team --> awaitingauthor,team: staged-recipes reviews
    awaitingauthor,team --> reviewrequested,team: User pings team OR rerequest review from member of staged recipes
```

**Comment by BastianZim** [comment]
We can definitely go with option two, my worries are that no-one uses the GitHub "Rerequest review tools" in staged-recipes. Based on my experience, most submitters aren't super familiar and just post a comment asking why no-one is reviewing their changes. I think I can count the times someone used the request review button on a PR I was reviewing on one hand. 😄 

Of course, we don't need to remove the "Awaiting author" label until the author requests a review again, that's just because sometimes someone might need some time to figure out a solution and this seems like rushing them but including that is not really a necessity. I would say that comes down to how much extra work it is including it but I don't need to have it, it can also be removed.

**Comment by carterbox** [comment]
> do you think that causes too much spam.

I dunno. Does GitHub send a separate notification for mention and review-request?

> [Awaiting author] not really a necessity

This is a good point. Is there really any difference between having the "awaiting author" tag and *not* having the "review requested" tag? I guess the only information that it adds is whether the PR has been reviewed at least once. Having the tag lets you look at the really old PR list and see if they were abandoned before or after they were reviewed.

I think we should go with the simplest case and then add complexity if someone says something.

```mermaid
stateDiagram-v2
    [*] --> reviewrequested,team: User pings team
    reviewrequested,team --> team: staged-recipes reviews
    team --> reviewrequested,team: User pings team
```

**Comment by BastianZim** [comment]
> Does GitHub send a separate notification for mention and review-request?

Yes, unfortunately.

> I think we should go with the simplest case and then add complexity if someone says something. 

Just to clarify, that would mean that we don't have any "Awaiting author" right? This was originally requested by core in https://github.com/conda-forge/staged-recipes/issues/18023#issuecomment-1038111489 so I think it would be good to have it included. The only difficulty there that I can see is creating a bot with the necessary rights. I tried to find them in the oauth settings but haven't been able to. Will be back in front of my computer on Monday though and will then check again.

**Comment by carterbox** [comment]
> Just to clarify, that would mean that we don't have any "Awaiting author" right?

Oh. I didn't realize it was specifically requested already. I can revert those commits that I made. I still think running this action for ANY author action (new commits / activity) is too much. I'd rather the author explicitly request review for the switch between request-review and awaiting-author tags to switch (the state diagram below).

```mermaid
stateDiagram-v2
    [*] --> reviewrequested,team: User pings team
    reviewrequested,team --> awaitingauthor,team: staged-recipes reviews
    awaitingauthor,team --> reviewrequested,team: User pings team OR rerequest review from member of staged recipes
```

**Comment by BastianZim** [comment]
Yeah if we can solve the authentication and I'll try that on Monday using GitHub secrets, this should be implementable using the flow that you have already added.

File: .github/workflows/automate-review-labels.yml [hunk 1]
```diff
@@ -54,23 +52,28 @@
...
-      name: 'When reviewed, add awaiting author label'
+    remove-review-requested-label:
+      name: 'When reviewed, remove review-requested label'
       if: >
-        ( 
+        (
           github.event.issue
           && github.event.issue.pull_request
           && (github.event.sender.login != github.event.issue.user.login)
           && contains(github.event.issue.labels.*.name, 'review-requested')
```
Comment (carterbox) [inline .github/workflows/automate-review-labels.yml:72]: I think this clause is only necessary if both normal comments AND review comments should count as a review.

Comment (BastianZim) [reply .github/workflows/automate-review-labels.yml:72]: True. I'll go through staged-recipes over the next few days and check what's the more common one.

**Comment by BastianZim** [comment]
I will close this for now as it seems that most steps are included in #11 now

**Pull request closed without merging**